Lib/fontTools/ttLib/tables/_g_l_y_f.py

```diff
@@ -1220,6 +1220,9 @@ def _ensureFloat(self):
 	def _checkFloat(self, p):
 		if self.isFloat():
 			return p
+		if any(v > 0x7FFF or v < -0x8000 for v in p):
```

Comment:
Merge this with the next if, so we only iterate once, and only if it matches, then check again, which case we have to deal with?  Maybe doesn't matter.

Reply:
the two checks are kind of independent from each other. The first if is to _ensureFloat whenever the value overflows a signed short, the second is to _ensureFloat when any value is actually a float. We want to check the first whether or not the type is float and short-circuit to _ensureFloat, without the extra casting to int and re-check for floats, because when we overflow it's all floats anyway.